int32 min_h
int32 max_h
---
int64 pos
